daemon/src/db.rs

```diff
@@ -408,88 +438,119 @@ pub async fn load_cfds_by_oracle_event_id(
-            let trading_pair = serde_json::from_str(row.trading_pair.as_str()).unwrap();
-            let position: Position = serde_json::from_str(row.position.as_str()).unwrap();
-            let price = serde_json::from_str(row.price.as_str()).unwrap();
-            let min_quantity = serde_json::from_str(row.min_quantity.as_str()).unwrap();
-            let max_quantity = serde_json::from_str(row.max_quantity.as_str()).unwrap();
-            let leverage = Leverage(row.leverage.try_into().unwrap());
-            let liquidation_price = serde_json::from_str(row.liquidation_price.as_str()).unwrap();
-            let creation_timestamp = serde_json::from_str(row.creation_timestamp.as_str()).unwrap();
-            let term = serde_json::from_str(row.term.as_str()).unwrap();
-            let origin: Origin = serde_json::from_str(row.origin.as_str()).unwrap();
-            let oracle_event_id = row.oracle_event_id.parse().unwrap();
-
-            let quantity = serde_json::from_str(row.quantity_usd.as_str()).unwrap();
-            let latest_state = serde_json::from_str(row.state.as_str()).unwrap();
-
             let order = Order {
-                id: order_id,
-                trading_pair,
-                position,
-                price,
-                min_quantity,
-                max_quantity,
-                leverage,
-                liquidation_price,
-                creation_timestamp,
-                term,
-                origin,
-                oracle_event_id,
+                id: row.uuid,
+                trading_pair: row.trading_pair,
+                position: row.position,
+                price: Usd(Decimal::from_str(&row.initial_price)?),
+                min_quantity: Usd(Decimal::from_str(&row.min_quantity)?),
+                max_quantity: Usd(Decimal::from_str(&row.max_quantity)?),
+                leverage: row.leverage,
+                liquidation_price: Usd(Decimal::from_str(&row.liquidation_price)?),
+                creation_timestamp: convert_to_system_time(row.ts_secs, row.ts_nanos)?,
+                term: Duration::new(row.term_secs, row.term_nanos),
+                origin: row.origin,
+                oracle_event_id: row.oracle_event_id.parse::<BitMexPriceEventId>()?,
             };
 
-            Cfd {
+            Ok(Cfd {
                 order,
-                quantity_usd: quantity,
-                state: latest_state,
-            }
+                quantity_usd: Usd(Decimal::from_str(&row.quantity_usd)?),
+                state: serde_json::from_str(row.state.as_str())?,
+            })
         })
-        .collect();
+        .collect::<Result<Vec<_>>>()?;
 
     Ok(cfds)
 }
 
+fn convert_to_system_time(row_secs: i64, row_nanos: i32) -> Result<SystemTime> {
+    let secs = row_secs as i128;
+    let nanos = row_nanos as i128;
+    let offset_dt = OffsetDateTime::from_unix_timestamp_nanos(secs * 1_000_000_000 + nanos)?;
+
+    Ok(SystemTime::from(offset_dt))
+}
```

Comment:
Can you create an issue to get rid of the nanoseconds for all these timestamps? Not worth the complexity IMO.